### PR TITLE
adding the group discovery for forGroups also into the plugin

### PR DESCRIPTION
**Pull request by tulinkry**
The authorization was not performed for subgroups or subprojects of a configured group for a plugin. The same code as it was in the stack needed to be in the plugin.

### CODE REVIEW

**Comment by vladak** [comment]
test case would be nice

File: test/org/opensolaris/opengrok/authorization/AuthorizationStackTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+         * <pre>
+         * G1 + P1
+         *    + P2
+         *    + P3
+         *    + G2
+         *       + P4
+         *       + P5
+         *       + P6
+         *       + P7
+         * G2 + P8
```
Comment (vladak) [inline test/org/opensolaris/opengrok/authorization/AuthorizationStackTest.java]: G3 ?

File: test/org/opensolaris/opengrok/authorization/AuthorizationStackTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import org.opensolaris.opengrok.configuration.Project;
+import org.opensolaris.opengrok.configuration.RuntimeEnvironment;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ *
+ * @author Krystof Tulinger
+ */
+public class AuthorizationStackTest {
```
Comment (vladak) [inline test/org/opensolaris/opengrok/authorization/AuthorizationStackTest.java]: maybe use more descriptive name ? `AuthorizationStackForGroupProjectTest` or such ?

**Comment by vladak** [comment]
besides the nits it looks good

File: test/org/opensolaris/opengrok/authorization/AuthorizationStackTest.java (hunk 2 above)
Comment (tulinkry) [inline test/org/opensolaris/opengrok/authorization/AuthorizationStackTest.java]: I changed the method name. This test name should correspond to the testing class AuthorizationStack

File: test/org/opensolaris/opengrok/authorization/AuthorizationStackTest.java (hunk 1 above)
Comment (tulinkry) [inline test/org/opensolaris/opengrok/authorization/AuthorizationStackTest.java]: typo, fixed

**Pull request merged**